File: Min Stack - Leetcode 155/Min Stack - Leetcode 155.py
class MinStack:
    def __init__(self):
        self.stack = []
        self.minStk = []
    
    def push(self, val: int) -> None:
        self.stack.append(val)
        if not self.minStk or self.minStk[-1] >= val:
            self.minStk.append(val)
        else:
            self.minStk.append(self.minStk[-1])

    def pop(self) -> None:
        self.stack.pop()
        self.minStk.pop()
    
    def top(self) -> int:
        return self.stack[-1]

    def getMin(self) -> int:
        return self.minStk[-1]

File: Min Stack - Leetcode 155/Min Stack - Leetcode 155.java
class MinStack {
    private Stack<Integer> stk;
    private Stack<Integer> minStk;

    public MinStack(){
        stk = new Stack<>();
        minStk = new Stack<>();
    }

    public void push(int val){
        stack.push(val);

        if (minStk.length === 0 || minStk.peek() >= val)
        {
            minStk.push(val);
        } else {
            minStk.push(minStk.peek());
        }
    }

    public void pop(){
        stk.pop();
        minStk.pop();
    }

    public int top(){
        return stk.peek();
    }

    public int getMin(){
        return minStk.peek();
    }
}